File: solutions/go/spiral-matrix/1/spiral_matrix.go
package spiralmatrix

func SpiralMatrix(size int) [][]int {
	matrix := make([][]int, size)

    for r := range matrix {
        matrix[r] = make([]int, size)
    }

	left := 0
    right := size - 1
    top := 0
    bottom := size - 1
    counter := 1

    for counter <= size * size {
            for x := left; x <= right; x++ {
                matrix[top][x] = counter
                counter++
            }
            top++
            for y := top ; y <= bottom; y++ {
                matrix[y][right] = counter
                counter++
            }
            right--
            for x := right ; x >= left; x-- {
                matrix[bottom][x] = counter
                counter++
            }
            bottom--
            for y := bottom; y >= top; y-- {
                matrix[y][left] = counter
                counter++
            }
            left++
    }
    
    return matrix
}

File: solutions/csharp/spiral-matrix/1/SpiralMatrix.cs
using System;

public class SpiralMatrix
{
    public static int[,] GetMatrix(int size)
    {
        var matrix = new int[size, size];

        int l = 0, r = size - 1, t = 0, b = size - 1;

        for (int counter = 1; counter <= size * size; )
        {
            for (int x = l; x <= r; ++x)
                matrix[t, x] = counter++;

            ++t;

            for (int y = t ; y <= b; ++y)
                matrix[y, r] = counter++;

            --r;

            for (int x = r ; x >= l; --x)
                matrix[b, x] = counter++;

            --b;

            for (int y = b; y >= t; --y)
                matrix[y, l] = counter++;

            ++l;
        }

        return matrix;
    }
}
